api/envoy/extensions/filter/udp/dns_filter/v3alpha/dns_filter.proto

```diff
@@ -44,9 +46,35 @@ message DnsFilterConfig {
     }
   }
 
+  // This message contains the configuration for the Dns Filter operating
+  // in a client context. This message will contain the timeouts, retry,
+  // and forwarding configuration for Envoy to make DNS requests to other
+  // resolvers
+  message ClientContextConfig {
+    option (udpa.annotations.versioning).previous_message_type =
+        "envoy.config.filter.udp.dns_filter.v2alpha.DnsFilterConfig.ClientContextConfig";
+
+    // Enable forwarding of the query to an upstream DNS server
```

Comment:
I'm a little confused on what this param does. Can you clarify? What is the behavior where one sets up a client context config and then does not forward?

Reply:
If forwarding is not enabled, none of the client fields will be used.  The client context is essentially controlling Envoy's behavior as a DNS client.

Reply:
Could that be indicated by the lack of the client context message itself?

Reply:
I plan on having the filter's external resolution timeout in the client context message.   That could be moved elsewhere if we eliminate the client context entirely.

I will remove the `forward_query` field and use `has_client_config` as its replacement